src/proxy/http/fallback.rs

```diff
@@ -0,0 +1,336 @@
+use bytes::Buf;
+use futures::{Async, Future, Poll};
+use http;
+use hyper::body::Payload;
+use proxy;
+use svc;
+
+use std::{fmt, marker::PhantomData, mem};
+
+#[derive(Debug)]
+pub struct Error<A> {
+    error: proxy::Error,
+    fallback: Option<http::Request<A>>,
+}
+
+pub type Layer<P, F, A> = MakeFallback<svc::Builder<P>, svc::Builder<F>, A>;
+
+#[derive(Debug)]
+pub struct MakeFallback<P, F, A> {
+    primary: P,
+    fallback: F,
+    _p: PhantomData<fn(A)>,
+}
+
+pub struct Service<P, F, A> {
+    primary: P,
+    fallback: F,
+    _p: PhantomData<fn(A)>,
+}
+
+pub struct MakeFuture<P, F, A>
+where
+    P: Future,
+    F: Future,
+{
+    primary: Making<P>,
+    fallback: Making<F>,
```

Comment:
What's going on here? Isn't this just Pending? Can the Service just hold pending services?

Reply:
This is less like `Pending` and more like `future::Join` --- the idea is that the future is complete when both the wrapped services complete. Reimplementing `join` felt like less work than making `join(...).map(...)` work w/o a closure